test/main.ts

```diff
@@ -980,18 +1056,14 @@ suite('IDBPObjectStore', () => {
   });
 
   test('createIndex', async () => {
-    const schemaDB = await openDBWithData();
-    db = schemaDB as IDBPDatabase;
-
-    const store1 = schemaDB.transaction('object-store').store;
-
-    typeAssert<
-      IsExact<Parameters<typeof store1.createIndex>[0], 'date' | 'title'>
-    >(true);
-
-    const store2 = db.transaction('object-store').store;
-
-    typeAssert<IsExact<Parameters<typeof store2.createIndex>[0], string>>(true);
+    db = (await openDB<TestDBSchema>(dbName, getNextVersion(), {
+      upgrade(db, oldVersion, newVersion, tx) {
+        const store = db.createObjectStore('object-store');
+        typeAssert<
+          IsExact<Parameters<typeof store.createIndex>[0], 'date' | 'title'>
+        >(true);
+      },
+    })) as IDBPDatabase;
```

Comment:
What's the reason for changing the shape of this test and removing part of it?

Reply:
This part is strange, I can't figure it out, why the type `Parameters<typeof store.createIndex>[0]` of `store1` and `store2` is different.

```ts
  test('createIndex', async () => {
    const schemaDB = await openDBWithData();
    db = schemaDB as IDBPDatabase;

    const store1 = schemaDB.transaction('object-store', 'versionchange').store;

    typeAssert<
      IsExact<Parameters<typeof store1.createIndex>[0], 'date' | 'title'>
    >(true);

    const store2 = db.transaction('object-store', 'versionchange').store;

    typeAssert<
      IsExact<Parameters<typeof store2.createIndex>[0], string>
    >(true);
  });
```

Reply:
Oh, I figured it out, `db = schemaDB as IDBPDatabase;`, here `db` has no `DBTypes`, so its type is different.

Reply:
Add this part back, since we can only use `createIndex` on the 'versionchange' objectStore created by `db.createObjectStore()`, I split the test to two. Even declare `db = schemaDB as IDBPDatabase;` within `upgrade()`, `db.transaction('object-store', 'versionchange').store;` still throw the error: `mode is not 'readonly' or 'readwrite'`